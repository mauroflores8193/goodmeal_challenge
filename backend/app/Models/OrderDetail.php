<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class OrderDetail extends Model {
    use HasFactory;

    protected $fillable = [
        'order_id',
        'product_id',
        'amount',
        'price',
    ];

    public function order() {
        return $this->belongsTo(Order::class);
    }

    public function product() {
        return $this->belongsTo(Product::class);
    }

    public function getNameAttribute() {
        return $this->product->name;
    }

    public function getTotalAttribute() {
        return $this->amount * $this->price;
    }
}
