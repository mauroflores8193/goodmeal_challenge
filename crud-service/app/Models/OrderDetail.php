<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class OrderDetail extends Model {
    use SoftDeletes, HasFactory;

    protected $fillable = [
        'order_id',
        'product_id',
        'amount',
        'price',
    ];

    public function order() {
        return $this->belongsTo(Order::class);
    }

    public function product() {
        return $this->hasMany(Product::class);
    }
}
