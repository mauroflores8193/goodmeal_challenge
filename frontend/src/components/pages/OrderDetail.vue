<template>
  <div class="bg-teal-50 px-6 h-full text-sm">
    <Header title="Detalle de la orden" url-back="/orders"/>
    <Card class="py-12">
      <h2 class="text-lg font-medium mb-4">{{ order.storeName }}</h2>
      <DescriptionItem v-for="(item, index) in infoItems" v-bind="item" :key="index"/>
      <hr class="my-4"/>
      <div class="font-medium">Productos</div>
      <div v-for="(detail, index) in order.details" :key="index" class="flex justify-between gap-2">
        <div>{{ detail.amount }}</div>
        <div>{{ detail.name }}</div>
        <div>{{ detail.total | formatPrice }}</div>
      </div>
      <hr class="my-4"/>
      <div class="flex justify-between" v-if="order.shippingCost > 0">
        <div>Delivery</div>
        <div>{{ order.shippingCost | formatPrice }}</div>
      </div>
      <div class="flex justify-between">
        <div class="font-medium">Monto total</div>
        <div class="font-medium">{{ order.total | formatPrice }}</div>
      </div>
    </Card>
  </div>
</template>

<script>
import DescriptionItem from "../molecules/DescriptionItem.vue"
import Header from "../molecules/Header.vue";
import Card from "../molecules/Card.vue";
import orderApi from "../../api/order-api";

export default {
  name: "OrderDetail",
  components: {
    Card,
    Header,
    DescriptionItem,
  },
  data: () => ({ order: {} }),
  async created() {
    this.order = await orderApi.get(this.$route.params.id)
  },
  computed: {
    deliveryTime() {
      return `${this.order.deliveryStarTime} - ${this.order.deliveryEndTime} hrs`
    },
    infoItems: function () {
      return [
        { title: 'Dirección:', description: this.order.storeAddress },
        { title: 'Horario de retiro:', description: this.deliveryTime },
        { title: 'Fecha de retiro:', description: this.order.deliveryDate },
        { title: 'N° de orden:', description: this.order.id },
      ]
    }
  }
}
</script>