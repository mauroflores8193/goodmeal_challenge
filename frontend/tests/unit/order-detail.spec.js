import { createLocalVue, mount } from '@vue/test-utils'
import VueRouter from 'vue-router'
import OrderDetail from '@/components/pages/OrderDetail.vue'
import flushPromises from 'flush-promises'
import axios from 'axios'
import order from './../mocks/order.json'

jest.mock('axios')

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()
test('OrderDetail.vue', async () => {
  const mockRoute = {
    params: {
      id: 1
    }
  }
  axios.get.mockResolvedValue({ data: order });
  const tree = mount(OrderDetail, {
    localVue, router, global: {      mocks: { $route: mockRoute }    }, computed: {
      emoticon() {
        return 'ðŸ¤¯'
      },
      bgColor() {
        return 'bg-pink-400'
      }
    }
  })
  await flushPromises();
  expect(tree).toMatchSnapshot();
})
