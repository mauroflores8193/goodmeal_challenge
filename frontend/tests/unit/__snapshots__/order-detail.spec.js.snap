// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetail.vue 1`] = `
<div class="bg-teal-50 px-6 h-full text-sm">
  <div class="h-16 flex relative items-center"><a href="#/orders" class="absolute left-0"></a>
    <h1 class="text-lg font-medium mx-auto">Detalle de la orden</h1>
  </div>
  <div class="card px-6 relative py-12">
    <div class="bg-teal-500 text-white w-fit px-4 py-1 text-sm rounded-b-2xl absolute top-0 right-6">Rescatada</div>
    <h2 class="text-lg font-medium mb-4">Chauncey Kihn</h2>
    <div class="flex gap-1">
      <div class="font-medium">Dirección:</div>
      <div>Qui sapiente rerum a vitae sint necessitatibus. Delectus incidunt consequatur ut eveniet tenetur et voluptatem.</div>
    </div>
    <div class="flex gap-1">
      <div class="font-medium">Horario de retiro:</div>
      <div>16:10 - 00:10 hrs</div>
    </div>
    <div class="flex gap-1">
      <div class="font-medium">Fecha de retiro:</div>
      <div>13/03/1986</div>
    </div>
    <div class="flex gap-1">
      <div class="font-medium">N° de orden:</div>
      <div>8</div>
    </div>
    <hr class="my-4">
    <div class="font-medium">Productos</div>
    <div class="flex justify-between gap-2">
      <div>2</div>
      <div>Wilma Fisher</div>
      <div>17406</div>
    </div>
    <hr class="my-4">
    <div class="flex justify-between">
      <div>Delivery</div>
      <div>4057</div>
    </div>
    <div class="flex justify-between">
      <div class="font-medium">Monto total</div>
      <div class="font-medium">17406</div>
    </div>
  </div>
</div>
`;
